pajbot/bot.py

```diff
@@ -851,17 +852,35 @@ def safe_say(self, message: str, channel: Optional[str] = None) -> None:
     def me(self, message: str, channel: Optional[str] = None) -> None:
         self.say("/me " + message[: CHARACTER_LIMIT - 4], channel=channel)
 
-    def announce(self, message: str, channel: Optional[str] = None) -> None:
+    def announce(self, message: str, channel_id: str = None) -> None:
         if message is None:
             log.warning("message=None passed to Bot::announce()")
             return
 
         if self.silent:
             return
 
+        if channel_id is None:
+            channel_id = self.streamer.id
+
         message = utils.clean_up_message(message)
 
-        self.privmsg("/announce " + message[: CHARACTER_LIMIT - 10], channel=channel)
+        try:
+            self.twitch_helix_api.send_chat_announcement(
+                self.bot_user.id,
+                {"message": message[: CHARACTER_LIMIT - 10]},
```

Comment:
```suggestion
                message[: CHARACTER_LIMIT - 10],
```

Reply:
`color` can also be in the body, so I'd like to leave that open for addition if anyone wishes to. Either that or I'll end up adding it as an optional field.

Reply:
I ended up adding it.